spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/ZuulRouteStore.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2013-2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.netflix.zuul.filters;
+
+import java.util.Map;
+
+/**
+ * An abstraction for retrieving the Zuul routes definitions.
+ *
+ * @author Jakub Narloch
+ */
+public interface ZuulRouteStore {
+
+    /**
+     * Retrieves the list of {@link org.springframework.cloud.netflix.zuul.filters.ZuulProperties.ZuulRoute}.
+     *
+     * @return list of routes
+     */
+    Map<String, ZuulProperties.ZuulRoute> getRoutes();
```

Comment:
Alright, but don't you think it maybe would make more sense to make ZuulRoute top level class? So that we could still use it ZuulProperties without need to use there ZuulRouteStore.ZuulRoute indirection.